exporter/sumologicexporter/config.go

```diff
@@ -92,10 +79,6 @@ func (cfg *Config) Validate() error {
 		return fmt.Errorf("invalid compression encoding type: %v", cfg.ClientConfig.Compression)
 	}
 
-	if cfg.CompressEncoding != NoCompression && cfg.ClientConfig.Compression != DefaultCompressEncoding {
-		return fmt.Errorf("compress_encoding is deprecated and should not be used when compression is set to a non-default value")
-	}
-
```

Comment:
I was thinking about migration and our previous approach to breaking changes and we should do it similar way like https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/sumologicexporter/config.go#L110-L111.

So instead of getting error that `compress_encoding` is not a correct configuration, we should return more infotmative message with details what should be done to apply the valid configuration.

I do not expect from the customers to read changelog, and it is better to repeat the message in the logs, which they definitely read when the collector is crashing (due to invalid configuration).

Following the above example there is difference between the following information:

```
support for the graphite metric format was removed, please use prometheus or otlp instead
```

and

```
unexpected metric format: graphite
```

Reply:
oh yeah this makes alot of sense, just to make sure i'm getting this right:

keep the `compress_encoding` field in the config struct, do something similar [like this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/sumologicexporter/config.go#L110-L111) and throw an better detailed error message saying something like `support for compress encoding has been deprecated, please use compression instead`

Reply:
Yup, thats the conception

I would change `deprecated` to `removed`